SpatialGISTPostgreSQL/SpatialPostgreSQLDatastore.cs

```diff
@@ -0,0 +1,232 @@
+using DynamicFlightStorageDTOs;
+using Npgsql;
+using Testcontainers.PostgreSql;
+
+namespace SpatialGISTPostgreSQL;
+
+public class SpatialPostgreSQLDatastore : IEventDataStore, IDisposable
+{
+    private PostgreSqlContainer? _container;
+    private NpgsqlConnection? _insertConnection;
+    private NpgsqlConnection? _updateConnection;
+    private readonly IWeatherService _weatherService;
+    private readonly IRecalculateFlightEventPublisher _flightRecalculation;
+    private readonly string _initScriptPath;
+    
+    public SpatialPostgreSQLDatastore(IWeatherService weatherService, IRecalculateFlightEventPublisher recalculateFlightEventPublisher)
+    {
+        var initScriptName = "gistInit.sql";
+        _weatherService = weatherService ?? throw new ArgumentNullException(nameof(weatherService));
+        _flightRecalculation = recalculateFlightEventPublisher ?? throw new ArgumentNullException(nameof(recalculateFlightEventPublisher));
+        _initScriptPath = Path.Combine(AppContext.BaseDirectory, "DatabaseInit", initScriptName ?? throw new ArgumentNullException(nameof(initScriptName)));
+        if (!File.Exists(_initScriptPath))
+        {
+            throw new ArgumentException("Init script not found at " + _initScriptPath, nameof(initScriptName));
+        }
+    }
+
+    public async Task StartAsync()
+    {
+        _container = new PostgreSqlBuilder().Build();

```

Comment:
Is this the same as this: https://testcontainers.com/modules/postgis/?language=dotnet ?